samples/bluetooth/central_bas/src/main.c

```diff
@@ -72,6 +74,21 @@ static void scan_connecting(struct bt_scan_device_info *device_info,
 BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL,
 		scan_connecting_error, scan_connecting);
 
+static void bas_c_read_cb(struct bt_gatt_bas_c *bas_c,
+				  u8_t battery_level,
+				  int err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(bt_gatt_bas_c_conn(bas_c)),
+			  addr, sizeof(addr));
+	if (err) {
+		printk("[%s] Battery read ERROR: %d\n", addr, err);
+		return;
+	}
+	printk("[%s] Battery read: %"PRIu8"%%\n", addr, battery_level);
+}
+
```

Comment:
We can get rid of this function by the usage of the notification one.

Reply:
We can not remove it because sample has implemented a characteristic value read on button pressing. This callback is used there